flowz/channels/tools.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import absolute_import
+from __future__ import print_function
+
+import sys
+
+
+class Roller(object):
+    def __init__(self, window_size):
+        assert window_size > 0
+        self.window_size = window_size
+        self.count = 0
+
+    def __call__(self, obj):
+        for i in range(self.count, self.count+self.window_size):
+            yield i
+        self.count += 1
+
+
+def rolling(window_size):
+    """
+    Return a callable that can be passed to windowby() to put channel objects
+    into rolling windows of `window_size`.
+
+    The return windows will ramp from size 1 up to `window_size`, then back down
+    to 1 by the end of the channel.  In order to constrain them to a particular
+    size, consider using `pin_group_size()`.
+    """
```

Comment:
I think it would be worthwhile to note in the docstring that the callable returned has internal state and therefore shouldn't be used across multiple cases if the user wants their app to make any damn sense.